### PR TITLE
DR-2292 Add script to rotate Azure secret

### PR DESCRIPTION
**Pull request by nmalfroy**
Document and provide a script to rotating the Azure TDR application secret

### CODE REVIEW

File: tools/rotate-azure-app-secret.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# This script takes in a new Azure application secret and applies it to all environments
+# To generate a secret:
+# - Log into the Azure portal (https://portal.azure.com/)
+# - Go to the DSP Terra Dev tenant
+# - Go to the App Registration service
+# - Click on the TDR application (https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Credentials/appId/22cb243c-f1a5-43d8-8f12-6566bcce6542/isMSAApp/)
```
Comment (tlangs) [inline tools/rotate-azure-app-secret.sh:7]: I cannot do this as I don't have permissions.

**Review by tlangs — CHANGES_REQUESTED** [review]

File: tools/rotate-azure-app-secret.sh (hunk 1 above)
Comment (nmalfroy) [reply tools/rotate-azure-app-secret.sh:7]: well, that's not good.  One sec

Comment (tlangs) [reply tools/rotate-azure-app-secret.sh:7]: Huzzah I can create a secret now!

File: tools/rotate-azure-app-secret.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Long running services (e.g. dev, alpha, staging, production) all need to be synced in Argo
+
+AZURE_APP_SECRET=$1
+
+update_secret () {
+  local location_name="$1"
+  local secret_location="$2"
+
+  echo "Updating secret for environment $location_name"
+  vault read --format=json $secret_location | jq --arg AZURE_APP_SECRET "$AZURE_APP_SECRET" '.data | .applicationsecret2=$AZURE_APP_SECRET' | vault write ${secret_location} -
```
Comment (tlangs) [inline tools/rotate-azure-app-secret.sh]: `vault read --format=json --field=data` will keep you from having to parse `.data` out in your `jq` query.

File: tools/rotate-azure-app-secret.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+
+update_secret "dev" "secret/dsde/datarepo/dev/helm-azure"
+update_secret "perf" "secret/dsde/datarepo/perf/helm-azure"
+update_secret "integration" "secret/dsde/datarepo/integration/helm-azure-integration"
+update_secret "alpha" "secret/dsde/datarepo/alpha/helm-azure"
+update_secret "staging" "secret/dsde/datarepo/staging/helm-azure"
+update_secret "production" "secret/dsde/datarepo/production/helm-azure"
```
Comment (tlangs) [inline tools/rotate-azure-app-secret.sh]: missing newline

File: tools/rotate-azure-app-secret.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  local secret_location="$2"
+
+  echo "Updating secret for environment $location_name"
+  vault read --format=json $secret_location | jq --arg AZURE_APP_SECRET "$AZURE_APP_SECRET" '.data | .applicationsecret2=$AZURE_APP_SECRET' | vault write ${secret_location} -
+}
+
+
+update_secret "dev" "secret/dsde/datarepo/dev/helm-azure"
+update_secret "perf" "secret/dsde/datarepo/perf/helm-azure"
+update_secret "integration" "secret/dsde/datarepo/integration/helm-azure-integration"
```
Comment (tlangs) [inline tools/rotate-azure-app-secret.sh:31]: the fact that integration is different from EVERY OTHER ENVIRONMENT is infuriating...

Comment (nmalfroy) [reply tools/rotate-azure-app-secret.sh:31]: yeah...I mean perf is too but not for this particular case

**Review by tlangs — APPROVED** [review]

**Review by samanehsan — APPROVED** [review]
Lgtm! :octocat:

**Pull request merged**